enclave/cert.c

```diff
@@ -327,19 +321,51 @@ static mbedtls_x509_crt* _sort_certs_by_issue_date(mbedtls_x509_crt* chain)
     return sorted;
 }
 
+/* Call mbedlts_x509_crt_verify and handle error logging */
+static oe_result_t _mbedtls_x509_crt_verify(
+    mbedtls_x509_crt* leaf_cert,
+    mbedtls_x509_crt* ca_cert_chain,
+    mbedtls_x509_crl* ca_crls)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    uint32_t flags = 0;
+    if (mbedtls_x509_crt_verify(
+            leaf_cert, ca_cert_chain, ca_crls, NULL, &flags, NULL, NULL) != 0)
+    {
+        char error[1024] = {0};
+        mbedtls_x509_crt_verify_info(error, sizeof(error), "", flags);
+        if (flags & MBEDTLS_X509_BADCERT_REVOKED)
+            result = OE_VERIFY_REVOKED;
+        else if (flags & MBEDTLS_X509_BADCRL_EXPIRED)
+            result = OE_VERIFY_CRL_EXPIRED;
+        else
+            result = OE_VERIFY_FAILED;
+
+        OE_RAISE_MSG(
+            result,
+            "mbedtls_x509_crt_verify failed with %s (flags=0x%x)\n",
```

Comment:
I might be helpful to also print the return value from the mbedtls function.

Reply:
Considered this, but opted to retain the existing behavior moved from `_verify_whole_chain`; mbedTLS headers do this unusual thing where they define their error codes as a _negation_ of a hex value, so you need to do an extra step just to create string value that's text searchable in the headers. The error string being returned is already more useful in that regard.

Reply:
Sounds good.